docs/src/modules/java-protobuf/pages/workflows.adoc

```diff
@@ -0,0 +1,200 @@
+
+
+IMPORTANT: For simplicity purposes, we are returning the internal state directly back to the requester. In a real-world scenario, it's usually better to instead convert this internal domain model into a public model so the internal representation is free to evolve without breaking clients code.
+
+// TODO uncomment in next PR A full transfer workflow source code is available https://github.com/lightbend/kalix-jvm-sdk/tree/main/samples/java-protobuf-transfer-workflow[here, {tab-icon}, window="new"]. Follow the `README` file to run and test it.
+
+== Pausing workflow
+
+A long-running workflow can be paused while waiting for some additional information to continue processing. A special `pause` transition can be used to inform Kalix that the execution of the Workflow should be postponed. By launching a request to a Workflow endpoint, the user can then resume the processing. Additionally, a Kalix Timer can be scheduled to automatically inform the Workflow that the expected time for the additional data has passed.
+
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow-compensation/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=pausing]
+----
+<1> Schedules a timer as a Workflow step action. Make sure that the timer name is unique for every Workflow instance.
+<2> Pauses the Workflow execution.
+
+NOTE: Remember to cancel the timer once the Workflow is resumed. Also, adjust the Workflow timeout to match the timer schedule.
+
+Exposing additional mutational endpoints from the Workflow implementation should be done with special caution. Accepting a request from such endpoints should only be possible when the Workflow is in the expected state.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow-compensation/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=resuming]
+----
+<1> Accepts the request only when status is `WAITING_FOR_ACCEPTATION`.
+<2> Otherwise, rejects the requests.
+
+== Error handling
+
+Design for failure is one of the key attributes of all Kalix components. Workflow has the richest set of configurations from all of them. It's essential to build robust and reliable solutions.
+
+=== Timeouts
+
+By default, a workflow run has no time limit. It can run forever, which in most cases is not desirable behavior. A workflow step, on the other hand, has a default timeout of 5 seconds timeout. Both settings can be overridden at the workflow definition level or for a specific step configuration.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow-compensation/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=timeouts]
+----
+<1> Sets a workflow global timeout.
+<2> Sets a default timeout for all workflow steps.
```

Comment:
Is the `global timeout` the total time the Workflow has to be completed? Meaning that the total sum of all the steps execution time has to be less than 5s. If so I would expect this Workflow only has 2 steps.

Reply:
Answering your question, yes. 5 sec is just a dummy value, so yes, only 2 slow steps.